Index: XPMEditorPanel.h
===================================================================
--- XPMEditorPanel.h	(revision 5)
+++ XPMEditorPanel.h	(working copy)
@@ -21,7 +21,7 @@
 class wxComboBox;
 class wxScrolledWindow;
 class wxStaticText;
-class XPMToggleBMPButton;
+class wxCustomButton;
 class wxSpinCtrl;
 class wxBoxSizer;
 class wxGridEvent;
@@ -130,40 +130,40 @@
 		void UpdateConfiguration(void); ///< @brief ask all the image editors to update their configuration
 
 		//(*Declarations(XPMEditorPanel)
-		XPMToggleBMPButton* RectangleButton;
-		XPMToggleBMPButton* PolygonButton;
+		wxCustomButton* RectangleButton;
+		wxCustomButton* PolygonButton;
 		wxBoxSizer* ToolSizer;
-		XPMToggleBMPButton* LassoButton;
+		wxCustomButton* LassoButton;
 		wxSpinCtrl* BMPHeight;
 		wxStaticText* sCursorPos;
 		wxScrolledWindow* DrawCanvas;
-		XPMToggleBMPButton* EraserButton;
+		wxCustomButton* EraserButton;
 		wxSpinCtrl* SpinCtrl1;
 		wxSpinCtrl* BMPWidth;
 		wxStaticText* StaticText2;
 		wxBoxSizer* CanvasSizer;
-		XPMToggleBMPButton* RRectButton;
+		wxCustomButton* RRectButton;
 		wxStaticText* StaticText6;
-		XPMToggleBMPButton* CurveButton;
+		wxCustomButton* CurveButton;
 		wxStaticText* StaticText1;
 		wxStaticText* StaticText3;
 		wxGrid* Grid1;
-		XPMToggleBMPButton* TextButton;
+		wxCustomButton* TextButton;
 		wxPanel* ToolPanel;
 		wxCheckBox* CheckBox1;
-		XPMToggleBMPButton* SelectButton;
+		wxCustomButton* SelectButton;
 		wxSpinCtrl* SpinCtrl3;
-		XPMToggleBMPButton* FillButton;
+		wxCustomButton* FillButton;
 		wxStaticText* StaticText5;
-		XPMToggleBMPButton* BrushButton;
+		wxCustomButton* BrushButton;
 		wxSpinCtrl* SpinCtrl2;
 		wxComboBox* ZoomFactor;
 		XPMColorPicker* ColourPicker;
 		wxBoxSizer* PanelSizer;
-		XPMToggleBMPButton* EllipseButton;
-		XPMToggleBMPButton* PenButton;
-		XPMToggleBMPButton* PipetteButton;
-		XPMToggleBMPButton* LineButton;
+		wxCustomButton* EllipseButton;
+		wxCustomButton* PenButton;
+		wxCustomButton* PipetteButton;
+		wxCustomButton* LineButton;
 		wxBoxSizer* ToolPanelSizer;
 		wxStaticText* StaticText4;
 		//*)
@@ -305,7 +305,7 @@
         XPMUndo *m_undo_buffer;  ///< \brief the Undo buffer
 
         //Drawing tools ids, & cursors
-        XPMToggleBMPButton* tools[XPM_NUMBER_TOOLS]; ///< \brief the buttons associated to the tools
+        wxCustomButton* tools[XPM_NUMBER_TOOLS]; ///< \brief the buttons associated to the tools
         int iToolUsed; ///< \brief the index of the tool currently in use
         wxCursor ToolCursor[XPM_NUMBER_TOOLS]; ///< \brief the cursors associated to the tools
         ToolData tdata; ///< \brief tool specific data
Index: XPMEditorPanel.cpp
===================================================================
--- XPMEditorPanel.cpp	(revision 5)
+++ XPMEditorPanel.cpp	(working copy)
@@ -31,7 +31,8 @@
 #include <wx/settings.h>
 #include <wx/grid.h>
 #include <wx/intl.h>
-#include "XPMToggleBMPButton.h"
+#include "wx/things/toggle.h"
+#include "wx/tglbtn.h"
 #include <wx/string.h>
 #include <wx/combobox.h>
 //*)
@@ -264,45 +265,45 @@
 	ToolPanel = new wxPanel(this, ID_PANEL1, wxDefaultPosition, wxDefaultSize, wxTAB_TRAVERSAL, _T("ID_PANEL1"));
 	ToolPanelSizer = new wxBoxSizer(wxVERTICAL);
 	BoxSizer6 = new wxBoxSizer(wxHORIZONTAL);
-	SelectButton = new XPMToggleBMPButton(ToolPanel,ID_SELECT_BUTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_SELECT_BUTN"));
+	SelectButton = new wxCustomButton(ToolPanel,ID_SELECT_BUTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer6->Add(SelectButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	LassoButton = new XPMToggleBMPButton(ToolPanel,ID_LASSO_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_LASSO_BTN"));
+	LassoButton = new wxCustomButton(ToolPanel,ID_LASSO_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer6->Add(LassoButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer6, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer5 = new wxBoxSizer(wxHORIZONTAL);
-	PenButton = new XPMToggleBMPButton(ToolPanel,ID_PEN_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_PEN_BTN"));
+	PenButton = new wxCustomButton(ToolPanel,ID_PEN_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer5->Add(PenButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	BrushButton = new XPMToggleBMPButton(ToolPanel,ID_BRUSH_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_BRUSH_BTN"));
+	BrushButton = new wxCustomButton(ToolPanel,ID_BRUSH_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer5->Add(BrushButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer5, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer7 = new wxBoxSizer(wxHORIZONTAL);
-	PipetteButton = new XPMToggleBMPButton(ToolPanel,ID_PIPETTE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_PIPETTE_BTN"));
+	PipetteButton = new wxCustomButton(ToolPanel,ID_PIPETTE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer7->Add(PipetteButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	FillButton = new XPMToggleBMPButton(ToolPanel,ID_FILL_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_FILL_BTN"));
+	FillButton = new wxCustomButton(ToolPanel,ID_FILL_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer7->Add(FillButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer7, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer10 = new wxBoxSizer(wxHORIZONTAL);
-	LineButton = new XPMToggleBMPButton(ToolPanel,ID_LINE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_LINE_BTN"));
+	LineButton = new wxCustomButton(ToolPanel,ID_LINE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer10->Add(LineButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	CurveButton = new XPMToggleBMPButton(ToolPanel,ID_CURVE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_CURVE_BTN"));
+	CurveButton = new wxCustomButton(ToolPanel,ID_CURVE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer10->Add(CurveButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer10, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer9 = new wxBoxSizer(wxHORIZONTAL);
-	EraserButton = new XPMToggleBMPButton(ToolPanel,ID_ERASER_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_ERASER_BTN"));
+	EraserButton = new wxCustomButton(ToolPanel,ID_ERASER_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer9->Add(EraserButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	TextButton = new XPMToggleBMPButton(ToolPanel,ID_TEXT_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_TEXT_BTN"));
+	TextButton = new wxCustomButton(ToolPanel,ID_TEXT_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer9->Add(TextButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer9, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer11 = new wxBoxSizer(wxHORIZONTAL);
-	RectangleButton = new XPMToggleBMPButton(ToolPanel,ID_RECTANGLE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_RECTANGLE_BTN"));
+	RectangleButton = new wxCustomButton(ToolPanel,ID_RECTANGLE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer11->Add(RectangleButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	PolygonButton = new XPMToggleBMPButton(ToolPanel,ID_POLYGON_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_POLYGON_BTN"));
+	PolygonButton = new wxCustomButton(ToolPanel,ID_POLYGON_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer11->Add(PolygonButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer11, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	BoxSizer8 = new wxBoxSizer(wxHORIZONTAL);
-	EllipseButton = new XPMToggleBMPButton(ToolPanel,ID_ELLIPSE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_ELLIPSE_BTN"));
+	EllipseButton = new wxCustomButton(ToolPanel,ID_ELLIPSE_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer8->Add(EllipseButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
-	RRectButton = new XPMToggleBMPButton(ToolPanel,ID_ROUNDEDRECT_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize,0,wxDefaultValidator,_T("ID_ROUNDEDRECT_BTN"));
+	RRectButton = new wxCustomButton(ToolPanel,ID_ROUNDEDRECT_BTN, wxNullBitmap,wxDefaultPosition,wxDefaultSize);
 	BoxSizer8->Add(RRectButton, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_CENTER_VERTICAL, 1);
 	ToolPanelSizer->Add(BoxSizer8, 0, wxALL|wxEXPAND|wxALIGN_LEFT|wxALIGN_BOTTOM, 0);
 	Grid1 = new wxGrid(ToolPanel, ID_GRID1, wxDefaultPosition, wxSize(16,100), wxSUNKEN_BORDER, _T("ID_GRID1"));
